### PR TITLE
Initial DIP process (based off Rust RFCs)

### PR DESCRIPTION
**Pull request by philpax**
Here we go! This is a modified version of the Rust RFC process that simplifies a few of the details as we don't have an ecosystem as complex as theirs yet.

Once this is merged, I'll start by converting one of the prospective-dips issues to a DIP so we can see what it's like to use.

### CODE REVIEW

**Review by goaaats — APPROVED** [review]
Looks good to me, thank you.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+- Additions to the `Dalamud` namespace.
+
+Some changes do not require a DIP:
+
+- Rephrasing, reorganizing, refactoring, or otherwise "changing shape does
+  not change meaning".
+- Additions that strictly improve objective, numerical quality criteria
+  (warning removal, speedup, better platform coverage, more parallelism, trap
+  more errors, etc.)
+- Additions only likely to be _noticed by_ other developers-of-Dalamud,
```
Comment (kalilistic) [inline README.md:56]: Wouldn't we use things for DIP like improving the merge process for plugins?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+[when you need to follow this process]: #when-you-need-to-follow-this-process
+
+You need to follow this process if you intend to make "substantial" changes to
+Dalamud, XIVLauncher, the surrounding infrastructure, or the DIP process itself.
+What constitutes a "substantial" change is evolving based on community norms and
+varies depending on what part of the ecosystem you are proposing to change, but
+may include the following.
+
+- Any semantic or syntactic change to Dalamud that is not a bugfix.
+- Removing Dalamud features, including those that are feature-gated.
```
Comment (kalilistic) [inline README.md:45]: What is "feature-gated"?

**Review by kalilistic — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (philpax) [reply README.md:45]: feature flags for features that aren't finalised yet, but I don't think we have any of those (yet?)

File: README.md (hunk 1 above)
Comment (philpax) [reply README.md:56]: that's fair, will nix that line

**Pull request merged**